tests/lib/structures/Josh.test.ts

```diff
@@ -21,4 +137,1083 @@ describe('Josh', () => {
+
+        const value = await josh[Method.Get]('test:inc');
+
+        expect(value).toBe(1);
+      });
+
+      test('GIVEN josh w/ number at path THEN incremented number at key and path', async () => {
+        await josh[Method.Set]('test:inc', 0, 'path');
+
+        const inced = await josh[Method.Inc]('test:inc', ['path']);
+
+        expect(inced).toBeInstanceOf(Josh);
+
+        const value = await josh[Method.Get]<{ path: number }>('test:inc');
+
+        expect(value ? value.path : 0).toBe(1);
+      });
+    });
+
+    describe(Method.Keys, () => {
+      test('GIVEN josh w/o data THEN returns data w/o data from keys', async () => {
+        const keys = await josh[Method.Keys]();
+
+        expect(keys).toEqual([]);
+      });
+
+      test('GIVEN josh w/ data THEN returns data from keys', async () => {
+        await josh[Method.Set]('test:keys', 'value');
+
+        const keys = await josh[Method.Keys]();
+
+        expect(keys).toEqual(['test:keys']);
+      });
+    });
+
+    describe(Method.Map, () => {
+      describe(Payload.Type.Hook, () => {
+        test('GIVEN josh w/o data THEN returns data w/o data', async () => {
+          const mapped = await josh[Method.Map]((value) => value);
+
+          expect(mapped).toEqual([]);
+        });
+
+        test('GIVEN josh w/ data THEN returns data', async () => {
+          await josh[Method.Set]('test:map', 'value');
+
+          const mapped = await josh[Method.Map]((value) => value);
+
+          expect(mapped).toEqual(['value']);
+        });
+      });
+
+      describe(Payload.Type.Path, () => {
+        test('GIVEN josh w/o data THEN returns data w/o data', async () => {
+          const mapped = await josh[Method.Map]([]);
+
+          expect(mapped).toEqual([]);
+        });
+
+        test('GIVEN josh w/ data THEN returns data', async () => {
+          await josh[Method.Set]('test:map', 'value');
+
+          const mapped = await josh[Method.Map]([]);
+
+          expect(mapped).toEqual(['value']);
+        });
+
+        test('GIVEN josh w/ data at path THEN returns data', async () => {
+          await josh[Method.Set]('test:map', 'value', 'path');
+
+          const mapped = await josh[Method.Map](['path']);
+
+          expect(mapped).toEqual(['value']);
+        });
+      });
+    });
+
+    describe(Method.Math, () => {
+      test('GIVEN josh w/ data THEN returns data', async () => {
+        await josh[Method.Set]('test:math', 0);
+
+        const mathed = await josh[Method.Math]('test:math', MathOperator.Addition, 1);
+
+        expect(mathed).toBeInstanceOf(Josh);
+
+        const value = await josh[Method.Get]('test:math');
+
+        expect(value).toBe(1);
+      });
+    });
+
+    describe(Method.Partition, () => {
+      describe(Payload.Type.Hook, () => {
+        test('GIVEN josh w/o data THEN returns data w/o data', async () => {
+          const parted = await josh[Method.Partition]((value) => value === 'value');
```

Comment:
could we have all these variables renamed to `result` or something else more consistent?